pkg/detectors/jdbc/sqlserver.go

```diff
@@ -38,22 +37,35 @@ func parseSqlServer(subname string) (jdbc, error) {
 		return nil, errors.New("expected connection to start with //")
 	}
 	conn := strings.TrimPrefix(subname, "//")
-	params := map[string]string{
-		"user id":  "sa",
-		"database": "master",
-	}
-	for _, param := range strings.Split(conn, ";") {
+
+	port := "1433"
+	var password, host string
+
+	for i, param := range strings.Split(conn, ";") {
 		key, value, found := strings.Cut(param, "=")
-		if !found {
+		if !found && i == 0 {
+			//  String connectionUrl = "jdbc:sqlserver://<server>:<port>;encrypt=true;databaseName=AdventureWorks;user=<user>;password=<password>";
+			if split := strings.Split(param, ":"); len(split) > 1 {
+				host = split[0]
+				port = split[1]
+			} else {
+				host = param
+			}
 			continue
 		}
-		params[key] = value
-		if key != "password" && strings.Contains(strings.ToLower(key), "password") {
-			params["password"] = value
+
+		switch strings.ToLower(key) {
+		case "password":
+			password = value
+		case "spring.datasource.password":
+			password = value
+		case "server":
+			host = value
+		case "port":
+			port = value
 		}
 	}
 	return &sqlServerJDBC{
-		conn:   strings.TrimPrefix(subname, "//"),
-		params: params,
+		connStr: fmt.Sprintf("sqlserver://sa:%s@%s:%s?database=master&connection+timeout=5", password, host, port),
```

Comment:
Does this mean we'll only ever try to connect to the `master` database for sqlserver?

Reply:
Yes, and sa for the user. This did not change. I would have improved it during this but the container did not allow them to be set. It is follow-up work.